### ISSUE TITLE
Create the RoadIntersection class and structure

### ISSUE THREAD

**Issue by TheDuckCow** [TheDuckCow/godot-road-generator#115]
Stepping stone to create the intersection support. This will have to cover:

- Allowing roadpoint children
- Having options to generate/refresh roadpoint children, so we can generate reference geometry in the scene without there being a road network
- This also means performing cleanup or propagating regeneration if needed from a RoadNetwork parent, should one exist.

**Comment by TheDuckCow** [comment]
Interesting [UX idea on intersections](https://www.gameuidatabase.com/index.php?scrn=46&scroll=224#&gid=1&pid=209), not sure it's really relevant since we'll inherently have a primary container for the intersection itself anyways. Could be interesting to see more about this game (Tropico) and how it handles UX especially since it's for mobile.

**Comment by TheDuckCow** [comment]
Here's [another video](https://www.youtube.com/watch?v=f4tBCwhBD34) (this time, an Unreal tool) showing intersections as just chunks you can place into the world. With its drag and drop nature, it seems like this one is actually quite nice to work with since it seems to not require/enforce very specific hierarchy structure (though the tools to actually form the connections aren't that great, all being done from side panels + picker tools back and forth).

**Comment by TheDuckCow** [comment]
Since I was having trouble finding this, uploading our old planning PDF which was talking about the concepts of how to define an intersection. Not suggesting this is still the mindset, just placing for historical reference.

[intersections v4.pdf](https://github.com/user-attachments/files/16924527/intersections.v4.pdf)

I'm still putting into question whether RoadIntersection needs to be its own user-facing class, as a special RoadPoint, or if a Container is just configured to be in an "intersection" mode to assume all of its open RoadPoints are intended to join together in an intersection. More thoughts to come in the future.

**Comment by TheDuckCow** [comment]
Yet another addon demo with some interesting features, and including a utility mode for affecting multiple intersections at once. Also shows another take for texturing / materials for side elements like sidewalks, which I'm increasingly realizing would be very nice to have.

**Comment by WinnerWind** [comment]
I would love to see this implemented, as I am making a city driving game, and a city without intersections is meaningless.
Godot 4.

**Comment by TheDuckCow** [comment]
The changes of https://github.com/TheDuckCow/godot-road-generator/pull/210/files and https://github.com/TheDuckCow/godot-road-generator/pull/212 will effectively unblock your needs here. Rather than establishing an actual intersection class in the meantime (which would be more useful for procedural generation and for specific traffic sim assumptions), the addon now just offers some hand-crafted, premade intersections along with Blender templates to create your own.

Procedural intersections are still in mind, we just don't have the bandwidth to prioritize that right now. Watch the repo as the above PR's are soon to be migrated into a godot 4 release.

**Comment by TheDuckCow** [comment]
Marking this as effectively done.
